projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -103,10 +118,25 @@ export class IgxCell {
             owner: this.grid
         };
         if (includeNewValue) {
-            args.newValue = this.editValue;
+            args.newValue = this.castToNumber(this.editValue);
         }
         return args;
     }
+
+    createDoneEditEventArgs(value: any): IGridEditDoneEventArgs {
+        const args: IGridEditDoneEventArgs = {
+            rowID: this.id.rowID,
+            cellID: this.id,
+            // rowData - should be the updated/committed rowData - this effectively should be the newValue
+            // the only case we use this.rowData directly, is when there is no rowEditing or transactions enabled
+            rowData: this.grid.transactions.enabled ? this.grid.transactions.getAggregatedValue(this.id.rowID, true) : this.rowData,
+            oldValue: this.value,
+            newValue: this.castToNumber(value),
```

Comment:
This cast might be redundant, since the value coming from the prev event has been case already and committed to data (unless `updateData` also casts). Otherwise, LGTM